Scripts/ShaderReloadSoakTest.bv.lua

```diff
@@ -0,0 +1,38 @@
+----------------------------------------------------------------------------------------------------
+--
+-- Copyright (c) Contributors to the Open 3D Engine Project.
+-- For complete copyright and license terms please see the LICENSE at the root of this distribution.
+--
+-- SPDX-License-Identifier: Apache-2.0 OR MIT
+--
+--
+--
+----------------------------------------------------------------------------------------------------
+
+-- WARNING: This is a soak test, do not add this test to the fully automated
+-- test suites.
+
+OpenSample('RPI/ShaderReloadSoakTest')
+ResizeViewport(500, 500)
+IdleSeconds(2)
+
+function TestButton(buttonName, shaderReloadWaitSeconds, pixelCaptureWaitSeconds)
+    SetImguiValue(buttonName, true)
+    IdleSeconds(shaderReloadWaitSeconds) -- Idle for a bit to give time for the shader to reload
+    SetShowImGui(false)
+    SetImguiValue('Check color', true)
+    IdleSeconds(pixelCaptureWaitSeconds)
+    SetShowImGui(true)
+end
+
+local shaderReloadWaitSeconds = 3
+local pixelCaptureWaitSeconds = 1
+
+
+-- Fixme. As a Soak Test, this should run for a long time and exit
+-- on the first failure, or upon user request.
+for i=1,5 do
+    TestButton("Red shader", shaderReloadWaitSeconds, pixelCaptureWaitSeconds)
```

Comment:
It's a very good test but how realistic is the assumption that it will always finish reload in 3 seconds?
Also, running 15 cycles of it seems like a long test (>15 seconds), is there away to bring it to 'fail conditions' and test it at that?

Reply:
I'm going to add some better notification trackings to reduce the wait time. The Lua API is insufficient to do what you mention, which I already thought about it too: 1- Ability to exit on first failure. 2- Ability to exit the test at any time.